src/audio/dai.c

```diff
@@ -64,6 +74,47 @@ struct dai_data {
 	uint64_t wallclock;	/* wall clock at stream start */
 };
 
+static void dai_atomic_trigger(void *arg, enum notify_id type, void *data);
+
+static struct dai_group *dai_add_group(struct comp_dev *dev, uint8_t group_id)
+{
+	struct list_item *dai_groups = &sof_get()->dai_groups;
+	struct dai_data *dd = comp_get_drvdata(dev);
+	struct list_item *group_item;
+	struct dai_group *group;
+
+	if (dd->group) {
+		comp_info(dev, "dai_add_group(), DAI already in group");
+		return dd->group;
```

Comment:
shouldn't this be an error? This would happen e.g. if the config IPC and hence the topology file provide bad data. Better to flag an error than silently go ahead and do things that were not intended.

Reply:
@plbossart the DAI config IPC is sent more than once especially in the case of HDA DAIs. So I'm not sure this should be an error.

Reply:
we could still check that the group_id is the same as configured previously.